gen/k5login_root

```diff
@@ -22,18 +22,17 @@ our $A_GROUP_DESTINATIONS;   *A_GROUP_DESTINATIONS =  \'urn:perun:group:attribut
 my $sortingFunction = getAttributeSorting($A_PRINCIPAL, 1);
 my %outputByDestination = ('all' => undef);
 
-my @resourcesData = $data->getChildElements;
-foreach my $resourceData (@resourcesData) {
+foreach my $resourceId ($data->getResourceIds()) {
 
-	foreach my $groupData (($resourceData->getChildElements)[0]->getChildElements) {
-		my %groupAttributes = attributesToHash $groupData->getAttributes;
+	foreach my $groupId ($data->getGroupIdsForResource( resource => "$resourceId" )) {
 		my @generatedDataDestination = ('all');
-		if(defined $groupAttributes{$A_GROUP_DESTINATIONS}) { @generatedDataDestination = @{$groupAttributes{$A_GROUP_DESTINATIONS}}; }
+		if(defined $data->getGroupAttributeValue(group => "$groupId", attrName => "$A_GROUP_DESTINATIONS" )) {
+			@generatedDataDestination = @{$data->getGroupAttributeValue(group => "$groupId", attrName => "$A_GROUP_DESTINATIONS" )};
+		}
 
-		foreach my $memberData (($groupData->getChildElements)[1]->getChildElements) {
-			my %memberAttributes = attributesToHash $memberData->getAttributes;
-			next if $memberAttributes{$A_USER_STATUS} ne 'VALID';
-			push @{$outputByDestination{$_}}, $memberAttributes{$A_PRINCIPAL} foreach @generatedDataDestination;
+		foreach my $memberId ($data->getMemberIdsForResource( resource => "$resourceId" )) {
```

Comment:
Only the group's members should be iterated here instead of all members on the resource:
`foreach my $memberId ($data->getMemberIdsForResourceAndGroup( resource => $resourceId, group => $groupId ))`

Reply:
Thanks, I fixed that.